gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java

```diff
@@ -529,9 +529,18 @@ protected void cleanup() {
    */
   private DataWriter<Object> buildWriter()
       throws IOException {
+    String writerId = this.taskId;
+
+    // Add the task starting time if configured.
+    // This is used to reduce file name collisions which can happen due to the scheduling of a task on multiple workers.
+    // One case where this occurs is when a worker gets disconnected from a Helix cluster.
```

Comment:
Doesn't it also happen if the task fails on a worker and is subsequently dispatched to another worker? 
File name collisions due to concurrent executions of the same task, or file name collisions due to serial executions of the same task.. both are equally bad.. right?